src/chainstate/stacks/db/blocks.rs

```diff
@@ -1511,8 +1636,45 @@ impl StacksChainState {
     }
 
     /// Do we have a particular microblock stream given it _indexed head microblock hash_?
-    pub fn has_confirmed_microblocks_indexed(&mut self, index_block_hash: &BlockHeaderHash) -> Result<bool, Error> {
-        StacksChainState::has_block_indexed(&self.blocks_path, index_block_hash)
+    pub fn has_confirmed_microblocks_indexed(&mut self, index_microblock_hash: &BlockHeaderHash) -> Result<bool, Error> {
+        StacksChainState::has_block_indexed(&self.blocks_path, index_microblock_hash)
+    }
+
+    /// How many microblocks are in a given stream?
+    pub fn get_microblock_stream_length(&self, index_anchor_block_hash: &BlockHeaderHash) -> Result<u64, Error> {
+        let sql = "SELECT COUNT(microblock_hash) FROM staging_microblocks WHERE index_block_hash = ?1".to_string();
```

Comment:
Can microblock-chains fork? Can you end up counting 2 micro-blocks chain forks, and returning the sum of the length of 2 forks?

Reply:
Good catch!  This code should filter by whether or not the microblock was also already processed (since the only place it gets used is to determine if there's a confirmed stream).